libraries/ui/src/ui/types/ContextAwareProfile.h

```diff
@@ -47,10 +49,18 @@ class ContextAwareProfile : public ContextAwareProfileParent {
     };
 
     ContextAwareProfile(QQmlContext* parent);
+    ~ContextAwareProfile();
+    void onIsRestrictedChanged(bool newValue);
+
     QQmlContext* _context{ nullptr };
-    bool _cachedValue{ false };
-    quint64 _cacheExpiry{ 0 };
-    constexpr static quint64 MAX_CACHE_AGE = MSECS_PER_SECOND;
+    std::atomic<bool> _isRestricted{ false };
+
+private:
+    typedef QSet<ContextAwareProfile*> ContextAwareProfileSet;
+    typedef QMap<QQmlContext*, ContextAwareProfileSet> ContextMap;
+
+    static QReadWriteLock gl_contextMapProtect;
+    static ContextMap gl_contextMap;
```

Comment:
I think dropping the `gl_` prefix would be more in line with the coding standard

Reply:
Almost tempted to raise an objection here -- does the coding standard have anything in it to handle global variables, and should it? Should these variables look the same as global variables? Should they be all caps?

Reply:
Actually one could argue that even though these are static, they are private, so they actually should start with an underscore: https://github.com/kasenvr/project-athena/blob/kasen/core/CODING_STANDARD.md#112-private-class-variables-must-be-in-mixed-case-prefixed-with-an-underscore

One could also argue that this rule implies global variables should NOT start with an underscore (although we actually don’t really follow this rule at all): https://github.com/kasenvr/project-athena/blob/kasen/core/CODING_STANDARD.md#119-global-variables-should-always-be-referred-to-using-the--operator

So...I’m open (but we don’t use gl_ anywhere that I’ve seen so I would vote against that)

Reply:
I guess if nothing else we could just go with a _global_ prefix